homeassistant/components/xiaomi_aqara.py

```diff
@@ -216,13 +232,37 @@ def device_state_attributes(self):
         """Return the state attributes."""
         return self._device_state_attributes
 
-    def push_data(self, data):
+    @callback
+    def _set_unavailable(self, now):
+        """Set state to UNAVAILABLE."""
+        self._remove_unavailability_tracker = None
+        self._is_available = False
+        self.schedule_update_ha_state()
+
+    def _track_unavailable(self):
+        if self._remove_unavailability_tracker:
+            self._remove_unavailability_tracker()
+        self._remove_unavailability_tracker = async_track_point_in_utc_time(
```

Comment:
This is an async method but I saw that you called it from the constructor, which is called from a sync context (`setup_platform`). This is not allowed.

Reply:
Now it's not running from constructor. Replaced with added_to_hass hook.